packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -264,59 +263,25 @@ export function linkify({
   blogPostsBySource,
   onBrokenMarkdownLink,
 }: LinkifyParams): string {
-  // TODO temporary, should consider the file being in localized folder!
-  const folderPath = contentPaths.contentPath;
-
-  let fencedBlock = false;
-  const lines = fileContent.split('\n').map((line) => {
-    if (line.trim().startsWith('```')) {
-      fencedBlock = !fencedBlock;
-    }
-
-    if (fencedBlock) {
-      return line;
-    }
-
-    let modifiedLine = line;
-    const mdRegex = /(?:(?:\]\()|(?:\]:\s?))(?!https)([^'")\]\s>]+\.mdx?)/g;
-    let mdMatch = mdRegex.exec(modifiedLine);
-
-    while (mdMatch !== null) {
-      const mdLink = mdMatch[1];
-
-      const aliasedSource = (source: string) =>
-        aliasedSitePath(source, siteDir);
-
-      const blogPost: BlogPost | undefined =
-        blogPostsBySource[aliasedSource(resolve(filePath, mdLink))] ||
-        blogPostsBySource[
-          aliasedSource(`${contentPaths.contentPathLocalized}/${mdLink}`)
-        ] ||
-        blogPostsBySource[
-          aliasedSource(`${contentPaths.contentPath}/${mdLink}`)
-        ];
-
-      if (blogPost) {
-        modifiedLine = modifiedLine.replace(
-          mdLink,
-          blogPost.metadata.permalink,
-        );
-      } else {
-        const brokenMarkdownLink: BlogBrokenMarkdownLink = {
-          folderPath,
-          filePath,
-          link: mdLink,
-        };
-        onBrokenMarkdownLink(brokenMarkdownLink);
-      }
-
-      mdMatch = mdRegex.exec(modifiedLine);
-    }
-
-    return modifiedLine;
-  });
-
-  return lines.join('\n');
+  return replaceMarkdownLinks(
+    fileContent,
+    filePath,
+    contentPaths,
+    {
+      siteDir,
+      sourceToPermalink: blogPostsBySource,
+      onBrokenMarkdownLink(data) {
+        onBrokenMarkdownLink({
+          // TODO: we should return here all contentPaths as we are checking them anyway
+          // TODO temporary, should consider the file being in localized folder!
+          folderPath: data.contentPaths.contentPath,
+          filePath: data.filePath,
+          link: data.link,
+        });
+      },
```

Comment:
> TODO temporary, should consider the file being in localized folder!

instead of passing `folderPath` se should update this code to use `contentPaths` and we can simply pass `onBrokenMarkdownLink` function here